scenario/BulkerScenario.ts

```diff
@@ -168,3 +169,94 @@ scenario(
+    const startingRewardBalance = await albert.getErc20Balance(rewardTokenAddress);
+    const rewardOwed = ((await rewards.callStatic.getRewardOwed(comet.address, albert.address)).owed).toBigInt();
+    const expectedFinalRewardBalance = collateralAssetAddress === rewardTokenAddress ?
+      startingRewardBalance + rewardOwed - toSupplyCollateral :
+      startingRewardBalance + rewardOwed;
+
+    // Albert's actions:
+    // 1. Supplies 10 units of collateral
+    // 2. Borrows 5 base
+    // 3. Transfers 2 base to Betty
+    // 4. Supplies 0.01 ETH
+    // 5. Withdraws 0.005 ETH
+    // 6. Claim rewards
+    const supplyAssetCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'address', 'uint'], [comet.address, albert.address, collateralAsset.address, toSupplyCollateral]);
+    const withdrawAssetCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'address', 'uint'], [comet.address, albert.address, baseAsset.address, toBorrowBase]);
+    const transferAssetCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'address', 'uint'], [comet.address, betty.address, baseAsset.address, toTransferBase]);
+    const supplyEthCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'uint'], [comet.address, albert.address, toSupplyEth]);
+    const withdrawEthCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'uint'], [comet.address, albert.address, toWithdrawEth]);
+    const claimRewardCalldata = utils.defaultAbiCoder.encode(['address', 'address', 'address', 'bool'], [comet.address, rewards.address, albert.address, true]);
+    const calldata = [
+      supplyAssetCalldata,
+      withdrawAssetCalldata,
+      transferAssetCalldata,
+      supplyEthCalldata,
+      withdrawEthCalldata,
+      claimRewardCalldata
+    ];
+    const actions = [
+      await bulker.ACTION_SUPPLY_ASSET(),
+      await bulker.ACTION_WITHDRAW_ASSET(),
+      await bulker.ACTION_TRANSFER_ASSET(),
+      await bulker.ACTION_SUPPLY_ETH(),
+      await bulker.ACTION_WITHDRAW_ETH(),
+      await bulker.ACTION_CLAIM_REWARD(),
+    ];
+    const txn = await albert.invoke({ actions, calldata }, { value: toSupplyEth });
+
+    // Final expectations
+    const baseIndexScale = (await comet.baseIndexScale()).toBigInt();
+    const baseSupplyIndex = (await comet.totalsBasic()).baseSupplyIndex.toBigInt();
+    const baseTransferred = getExpectedBaseBalance(toTransferBase, baseIndexScale, baseSupplyIndex);
+    expect(await comet.collateralBalanceOf(albert.address, collateralAsset.address)).to.be.equal(toSupplyCollateral);
+    expect(await baseAsset.balanceOf(albert.address)).to.be.equal(toBorrowBase);
+    expect(await comet.balanceOf(betty.address)).to.be.equal(baseTransferred);
+    // NOTE: differs from the equivalent scenario for non-ETH markets
+    expect(await comet.borrowBalanceOf(albert.address)).to.be.equal(toBorrowBase + toTransferBase - (toSupplyEth - toWithdrawEth));
```

Comment:
Just to make sure I understand, if the scenario changed the collateral asset from ETH to something else it would be the same right?

Reply:
The `non-WETH` version of this test has these two assertions instead:

```
expect(await comet.borrowBalanceOf(albert.address)).to.be.equal(toBorrowBase + toTransferBase);
expect(await comet.collateralBalanceOf(albert.address, WETH.address)).to.be.equal(toSupplyEth - toWithdrawEth);
```

So in a USDC deployment, the borrow balance is how much the user borrowed and transferred. As well, the user has a collateral balance of WETH equal to how much ETH they supplied minus how much they borrowed.

But in this case, since the base is WETH, then the borrow balance is how much WETH was borrowed, how much WETH was transferred, how much ETH was borrowed (minus how much ETH was supplied)